src/odemis/driver/lakeshore.py

```diff
@@ -344,98 +403,234 @@ def HeaterSetup(self,
         except:
             # another exception.
             logging.exception("Failed to read sensor temperature.")
 
 
+class SerialBusAccesser(object):
+    """
+    Manages serial bus connections
+    """
+
+    def __init__(self, port: str):
+        """
+        port: the full path to the serial port (eg: /dev/ttyUSB1)
+        raises:
+           IOError: if something went wrong opening the port
+        """
+        # Ensure no one will talk to it simultaneously, and we don't talk to devices already in use
+        self._ser_access = threading.Lock()
+
+        # For debugging purpose
+        if port == "/dev/fake":
+            self._serial = LakeshoreSimulator(timeout=1)
+            self._file = None
+
+            return
+
+        self._file = open(port)  # Open in RO, just to check for lock
+        fcntl.flock(self._file.fileno(), fcntl.LOCK_EX | fcntl.LOCK_NB)  # Raises IOError if cannot lock
+
+        self._serial = self._openSerialPort(port)
+
+    def terminate(self):
+        with self._ser_access:
+            self._serial.close()
+
+    def sendOrder(self, cmd):
+        """
+        cmd (byte str): command to be sent to device (without the LF)
+        """
+        cmd = cmd + b"\n"
+        with self._ser_access:
+            logging.debug("Sending command %s", to_str_escape(cmd))
+            self._serial.write(cmd)
+            time.sleep(0.05)
+
+    def sendQuery(self, cmd):
+        """
+        cmd (byte str): command to be sent to device (without the LF, but with the ?)
+        returns (byte str): answer received from the device (without \n or \r)
+        raise:
+            IOError if no answer is returned in time
+        """
+
+        cmd = cmd + b"\n"
+        with self._ser_access:
+            logging.debug("Sending command %s", to_str_escape(cmd))
+            self._serial.write(cmd)
+
+            self._serial.timeout = 1
+            ans = b''
+            while ans[-1:] != b'\n':
+                char = self._serial.read()
+                if not char:
+                    raise IOError("Timeout after receiving %s" % to_str_escape(ans))
+                ans += char
+
+            logging.debug("Received answer %s", to_str_escape(ans))
+
+            return ans.strip()
+
+
+class IPBusAccesser(object):
+    """
+    Manages ip bus connections through ethernet connection
+    """
+
+    def __init__(self, ip_address: str):
+        """
+        ip_address (str): IP address to establish the ethernet connection
+        """
+        self._socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self._socket.settimeout(3)  # set 3 seconds timeout to receive data
+        try:
+            self._socket.connect((ip_address, MODEL_350_TCP_PORT))
+        except socket.timeout:
+            raise model.HwError("Connection is timed out.")
+
+        self._ser_access = threading.Lock()
+
+        # self._socket.recv.timeout(3)  # set 3 seconds timeout to receive data
```

Comment:
why is this commented?

Reply:
That line of code moved a couple of lines above, i'll remove it completely.